### PR TITLE
Feature/adding functionality to developer dashboard

### PR DESCRIPTION
**Pull request by sajidali36**

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -5,11 +5,24 @@ class Ability
 
   def initialize(user)
     user ||= User.new # guest user (not logged in)
-    if user.user_type.zeros?
+    # case user
+    # when 'manager'
+    #   can :manage, :all
+    # when 'developer'
```
Comment (MIbtehajNasar) [inline app/models/ability.rb]: can we remove this commented code if not needed?

File: app/models/user.rb [hunk 2]
```diff
@@ -6,6 +6,12 @@ class User < ApplicationRecord
 
   has_many :bugs_users
   has_many :bugs, through: :bugs_users
+
+  enum user_type: {
+    manager: 0,
```
Comment (MIbtehajNasar) [inline app/models/user.rb:11]: did you make corresponding changes across the application for this?

File: app/views/dashboard_developers/index.html.erb [hunk 3]
```diff
@@ -1,2 +1,343 @@
...
+                        <th>Description</th>
+                        <th>Show</th>
+                    </tr>
+                </thead>
+                <tbody>
+                  <% @projects.each do |project| %>
+                    <tr>
+                      <td class="w-25 p-3"><%= project.title %></td>
+                      <td><%= project.description %></td>
+                      <td><%= link_to 'Show', project, class: "show" %></td>
```
Comment (MIbtehajNasar) [inline app/views/dashboard_developers/index.html.erb]: kindly use project_path here

**Review by MIbtehajNasar — CHANGES_REQUESTED** [review]

File: app/controllers/dashboard_developers_controller.rb [hunk 4]
```diff
@@ -1,5 +1,8 @@
 # frozen_string_literal: true
 
 class DashboardDevelopersController < ApplicationController
-  def index; end
+  authorize_resource :class => :DashboardDevelopersController
```
Comment (MIbtehajNasar) [inline app/controllers/dashboard_developers_controller.rb]: kindly use the updated syntax for hash `class: :DashboardDevelopersController`

**Review by MIbtehajNasar — CHANGES_REQUESTED** [review]

File: app/controllers/dashboard_developers_controller.rb (hunk 4 above)
Comment (sajidali36) [reply app/controllers/dashboard_developers_controller.rb]: please review code again.

File: app/views/projects/_form_for _adding_users.html.erb [hunk 5]
```diff
@@ -26,7 +26,7 @@
     <div class="col-lg-8 col-lg-offset-4">
         <h4>all users</h4>
         <div class="input-group col-12">
-          <%= form.select(:user_ids, User.where.not(user_type: 0).collect {|u| [u.name, u.id]}, { include_blank: false, include_hidden: false }, {multiple: true}) %>
+          <%= form.select(:user_ids, User.where.not(user_type: 'manager').collect {|u| [u.name, u.id]}, { include_blank: false, include_hidden: false }, {multiple: true}) %>
```
Comment (MIbtehajNasar) [inline app/views/projects/_form_for _adding_users.html.erb:29]: prefer using symbols over a string, `:manager`

**Review by MIbtehajNasar — APPROVED** [review]

**Pull request merged**